client/src/BidDetail.js

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { get } from "./TenderClient";
+
+const BidDetail = () => {
+	const [bid, setBid] = useState({});
+	const [errorMsg, setErrorMsg] = useState(null);
+	const [loading, setLoading] = useState(true);
+	const { bidId } = useParams();
+
+	const fetchBid = async () => {
+		setLoading(true);
+		try {
+			const data = await get(`/api/bid-detail/${bidId}`);
+			setBid(data);
+			setErrorMsg(null);
+		} catch (error) {
+			setErrorMsg("Error fetching bid");
+		} finally {
+			setLoading(false);
+		}
+	};
+
+	useEffect(
+		() => {
+			fetchBid();
+		},
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[bidId]
+	);
```

Comment:
```suggestion
	useEffect(
		() => {
			fetchBid(bidId);
		},
		[bidId]
	);
```

Reply:
I have tried  but still have formate error.

Reply:
What error?

Reply:
![Screenshot from 2024-08-09 15-27-50](https://github.com/user-attachments/assets/a8c5dd83-e6a4-4da5-8cca-a14a0d6793b1)

Reply:
The answer is right there in the error message

Reply:
I fixed it now.